### PR TITLE
added edit user page

### PR DESCRIPTION
**Pull request by ihsanahmed-tpx**
### Context

We need to be able to amend a user's name and/or roles after they've been created.

### Changes proposed in this pull request

- added a new edit page
- added test to match the code change

### Checklist

-   [#1170]

### CODE REVIEW

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Events/UserEditedEvent.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+namespace TeachingRecordSystem.Core.Events;
+
+public record UserEditedEvent : EventBase
+{
+    public required User User { get; init; }
+    public required Guid AddedByUserId { get; init; }
+}
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.Core/Events/UserEditedEvent.cs:7]: Need a `Changes` enum here too (like we use in ID) to track what actually changed

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/EditUser.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        }
+
+        if (!ModelState.IsValid)
+        {
+            return this.PageWithErrors();
+        }
+
+        var user = await _dbContext.Users.SingleOrDefaultAsync(u => u.UserId == UserId);
+
+        var rolesChanged = Roles!.SequenceEqual(user!.Roles);
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/EditUser.cshtml.cs:68]:
```suggestion
        var rolesChanged = Roles!.SequenceEqual(user.Roles);
```

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Events/UserEditedEvent.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace TeachingRecordSystem.Core.Events;
+
+public record UserEditedEvent : EventBase
```
[the remark this answers is not in the document]
Comment (ihsanahmed-tpx) [reply TeachingRecordSystem/src/TeachingRecordSystem.Core/Events/UserEditedEvent.cs]: renamed to UserUpdatedEvent

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/EditUser.cshtml.cs (hunk 2 above)
Comment (ihsanahmed-tpx) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/EditUser.cshtml.cs]: This check has been removed from the code

**Review by gunndabad — APPROVED** [review]

**Pull request merged**